test/json/test-functions.json

```diff
@@ -0,0 +1,259 @@
+      "params": { "foo": 1, "bar": 1 },
+      "exp": "one one"
+    },
+    {
+      "src": ".match {$foo :number} {$bar :number} one one {{one one}} one * {{one other}} * * {{other}}",
+      "params": { "foo": 1, "bar": 2 },
+      "exp": "one other"
+    },
+    {
+      "src": ".match {$foo :number} {$bar :number} one one {{one one}} one * {{one other}} * * {{other}}",
+      "params": { "foo": 2, "bar": 2 },
+      "exp": "other"
+    },
+    {
+      "src": ".input {$foo :number} .match {$foo} one {{one}} * {{other}}",
+      "params": { "foo": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".local $foo = {$bar :number} .match {$foo} one {{one}} * {{other}}",
+      "params": { "bar": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".input {$foo :number} .local $bar = {$foo} .match {$bar} one {{one}} * {{other}}",
+      "params": { "foo": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".input {$bar :number} .match {$bar} one {{one}} * {{other}}",
+      "params": { "bar": 2 },
+      "exp": "other"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": { "bar": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": { "bar": 2 },
+      "exp": "other"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": { "bar": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": { "bar": 2 },
+      "exp": "other"
+    },
+    {
+      "src": ".input {$none} .match {$foo :number} one {{one}} * {{{$none}}}",
+      "params": { "foo": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".local $bar = {$none} .match {$foo :number} one {{one}} * {{{$bar}}}",
+      "params": { "foo": 1 },
+      "exp": "one"
+    },
+    {
+      "src": ".local $bar = {$none} .match {$foo :number} one {{one}} * {{{$bar}}}",
+      "params": { "foo": 2 },
+      "exp": "{$none}",
+      "errors": [{ "type": "unresolved-var" }]
+    },
+    {
+      "src": "{42 :number @foo @bar=13}",
+      "exp": "42",
+      "parts": [
+        { "type": "number", "parts": [{ "type": "integer", "value": "42" }] }
+      ]
+    }
+  ],
+  "ordinal": [
+    {
+      "src": ".match {$foo :ordinal} one {{st}} two {{nd}} few {{rd}} * {{th}}",
+      "params": { "foo": 1 },
+      "exp": "st"
+    },
+    {
+      "src": ".match {$foo :ordinal} one {{st}} two {{nd}} few {{rd}} * {{th}}",
+      "params": { "foo": 2 },
+      "exp": "nd"
+    },
+    {
+      "src": ".match {$foo :ordinal} one {{st}} two {{nd}} few {{rd}} * {{th}}",
+      "params": { "foo": 3 },
+      "exp": "rd"
+    },
+    {
+      "src": ".match {$foo :ordinal} one {{st}} two {{nd}} few {{rd}} * {{th}}",
+      "params": { "foo": 4 },
+      "exp": "th"
+    },
```

Comment:
Do you have the ability to reference ICU RBNF functionality?  ICU MessageFormat could reference them as the spellout and ordinal, which conflicts with this plural rule ordinal.

See here for reference: https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classMessageFormat.html

Please ignore duration from MessageFormat. It's deprecated, and you need a decent message formatting framework to do it right.

Reply:
You're missing the point of this test, which is testing the `ordinal` selector (== `selectordinal` in MF1). @eemeli  has used the English ordinal suffixes in the test for readability in a test that is certainly not ready for any locale other than en-US.

Note well that we are considering spellout, ordinal, and duration for inclusion into the default registry as formatters--in another set of issues and design documents. I recommended against including spellout and duration in the default registry. `ordinal` as a formatter is on the bubble (only ICU4J has it, not Intl or ICU4X) and there are obvious functionality problems with ordinals in the current implementation.